app/src/main/java/io/homeassistant/companion/android/util/compose/ModalBottomSheet.kt

```diff
@@ -0,0 +1,66 @@
+package io.homeassistant.companion.android.util.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.dp
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.common.R as commonR
+
+/**
+ * A Material 3-style modal bottom sheet with an optional handle, for use with a
+ * [com.google.android.material.bottomsheet.BottomSheetDialogFragment].
+ */
+@Composable
+fun ModalBottomSheet(
```

Comment:
I know this in alpha but whenever we upgrade we should use the actual library: https://developer.android.com/jetpack/androidx/releases/compose-material3#1.1.0-alpha06

Reply:
Material 3 doesn't really play nice with Material 2 which we currently use... that's another project for the future :)